rs/hang-cli/src/client.rs

```diff
@@ -18,32 +21,39 @@ pub async fn client<T: AsyncRead + Unpin>(
 	tracing::info!(%url, %name, "connecting");
 	let session = client.connect(url).await?;
 
-	// Create an origin producer to publish to the broadcast.
 	let origin = moq_lite::Origin::produce();
-
-	// Establish the connection, not providing a subscriber.
 	let session = moq_lite::Session::connect(session, origin.consumer, None).await?;
-
-	let mut import = Import::new(broadcast.producer.into(), format);
-	import
-		.init_from(input)
-		.await
-		.context("failed to initialize from media stream")?;
-
-	// Announce the broadcast as available once the catalog is ready.
 	origin.producer.publish_broadcast(&name, broadcast.consumer);
 
-	// Notify systemd that we're ready.
 	let _ = sd_notify::notify(true, &[sd_notify::NotifyState::Ready]);
```

Comment:
Can you try to call this only after `init_from`? It's nice because it means `systemctl start` will block until the media is initialized. Ideally the same for `publish_broadcast` but that doesn't matter as much.